gallery/src/components/NxToast/NxToastComplexLayoutExample.tsx

```diff
@@ -0,0 +1,166 @@
+  NxFormGroup
+} from '@sonatype/react-shared-components';
+import {
+  faArrowLeft,
+  faArrowRight,
+  faLink
+} from '@fortawesome/free-solid-svg-icons';
+
+const sidebarLogoPath = require('../../assets/images/logo-plaid-villain-text.png');
+
+interface ToastModel {
+  id: number;
+  alertComponent: ComponentType;
+  message: string;
+}
+
+export default function NxToastComplexLayoutExample() {
+  const [toastIdInc, setToastIdInc] = useState<number>(0);
+  const [toasts, setToasts] = useState<ToastModel[]>([]);
+
+  const addToast = (alertComponent: ComponentType, message: string) => {
+    const toastId = toastIdInc + 1;
+    setToastIdInc(toastId);
+    setToasts([
+      { id: toastId, alertComponent, message },
+      ...toasts
+    ]);
+  };
+
+  const removeToast = (id: number) => setToasts(reject(propEq('id', id), toasts));
+
+  return (
+    <>
+      <header className="nx-global-header">
+        <NxBackButton href="#/pages/Toast" targetPageTitle="Documentation" />
+      </header>
+      <NxToastContainer>
+        {
+          toasts.map(({ id, alertComponent, message }) => {
+            const ToastAlert = alertComponent;
+            return (
+              <NxToast key={id}
+                       onClose={()=> removeToast(id)}>
+                <ToastAlert>{message}</ToastAlert>
+              </NxToast>
+            );
+          })
+        }
+      </NxToastContainer>
+      <NxStatefulGlobalSidebar isDefaultOpen={false}
+                               toggleOpenIcon={faArrowLeft}
+                               toggleCloseIcon={faArrowRight}
+                               logoImg={sidebarLogoPath}
+                               logoAltText="RSC Plaid Villain"
+                               logoLink="#">
+        <NxGlobalSidebarNavigation>
+          <NxGlobalSidebarNavigationLink icon={faLink} text="NxGlobalSidebar" href="#/pages/NxGlobalSidebar"/>
+        </NxGlobalSidebarNavigation>
+      </NxStatefulGlobalSidebar>
+      <NxPageMain>
+        <NxPageTitle>
+          <NxH1>Lorem Ipsum</NxH1>
+        </NxPageTitle>
+        <NxTile>
+          <NxTile.Header>Scroll Down to Find Toast Buttons</NxTile.Header>
+          <NxTile.Content>
+            <NxP>
+              Loresssm ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore
+              dolore magna aliqua. Tortor consequat id porta nibh venenatis cras. Proin libero nunc consequat interdu
+              varius sit amet mattis. Enim praesent elementum facilisis leo vel. A arcu cursus vitae congue mauris
+              rhoncus aenean vel elit. Morbi tincidunt ornare massa eget egestas. Semper auctor neque vitae tempus qu
+              pellentesque nec nam aliquam. Faucibus nisl tincidunt eget nullam non nisi est sit. Sit amet aliquam id
+              diam maecenas.
+            </NxP>
+            <NxFormGroup label ="Pre-filled Input">
+              <NxTextInput isPristine={true} value="can't change this"/>
```

Comment:
Let's make it an `NxStatefulTextInput`. That way it's not weirdly frozen but you still don't need to add a bunch of logic around it.

Reply:
> Let's make it an `NxStatefulTextInput`. That way it's not weirdly frozen but you still don't need to add a bunch of logic around it.

Much better. Implemented here: [20438bf](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/20438bf0271d1ca2268542725abbaf6b771595e4)